x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/create_analytics_flyout/create_analytics_flyout.tsx

```diff
@@ -44,7 +44,7 @@ export const CreateAnalyticsFlyout: FC<CreateAnalyticsFormProps> = ({
         {(!isJobCreated || !isJobStarted) && (
           <EuiButtonEmpty onClick={closeModal}>
             {i18n.translate('xpack.ml.dataframe.analytics.create.flyoutCancelButton', {
-              defaultMessage: 'Cancel',
+              defaultMessage: isJobCreated === true ? 'Close' : 'Cancel',
```

Comment:
just questioning having logic in the `defaultMessage`
is this allowed? which will be translated?

Reply:
Nope - good catch. Updated 👍